pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java

```diff
@@ -110,6 +115,23 @@ private static boolean argsContains(String[] args, String arg) {
         return Arrays.asList(args).contains(arg);
     }
 
+    private static ClusterData retrieveClusterData(ServiceConfiguration config) throws Exception {
+        String configurationStoreServers = config.getConfigurationStoreServers();
+        Long zkSessionTimeoutMillis = config.getZooKeeperSessionTimeoutMillis();
+        String clusterName = config.getClusterName();
+        ZooKeeperClientFactory zkfactory = new ZookeeperClientFactoryImpl();
+        ZooKeeper configStoreZk = zkfactory.create(
+            configurationStoreServers, ZooKeeperClientFactory.SessionType.ReadWrite, zkSessionTimeoutMillis.intValue()).get();
+        byte[] metadata = configStoreZk.getData("/admin/clusters/" + clusterName, false, null);
+        ClusterData clusterData = ObjectMapperFactory.getThreadLocal().readValue(metadata, ClusterData.class);
+        return clusterData;
+    }
+
+    private static int retrievePort(String url) {
+        String[] uris = url.split(":");
+        return Integer.parseInt(uris[uris.length-1]);
```

Comment:
This can be problematic, since the url might potentially contain multiple hosts. since we introduced multiple hosts support in service url since 2.3.0. I would suggest using ServiceURI for parsing the service url and get the list of hosts in the service url.

Reply:
thanks for the help.